### ISSUE TITLE
qgis: postgres and postgis

### ISSUE THREAD

**Issue by miguelangelperg** [OSGeo/homebrew-osgeo4mac#712]
Hi @fjperini  
I would like to ask two questions related to _postgres_ and _postgis_. I imagine there will be powerful reasons, but:
1. If the user has a postgresql server installed through homebrew, why is it necessary to update the postgresql server when updating Qgis? It would not be possible that postgresql would only be installed if the user does not have it?
2. Why postgis 2.5.1_1 is installed in _usr/local/Cellar/postgis2_?
Is there any reason not to install it following the previous logic where in _usr/local/Cellar/postgis/_ you could find folders with different versions: _2.4.4_1_,  _2.5.0_, ....?
These questions were raised after a previous experience in which after running `brew upgrade qgis3`, my postgresql server was updated from version 10 to 11. The result is that my data cluster, about 168GB, was not accessible to the new server. This does not always have an easy solution. For example, using `brew postgresql-upgrade-database` successfully requires that both versions of the server have the same extensions, and initialized under the same encoding. And this does not always happen. Thus, my old data cluster contains numerous tables with postgis, while the new version of _postgresql11_ installed is started without _postgis_. The same happens with the _postgresql@10_  version that is installed by the command `brew postgresql-upgrade-database`. This is a cause for **pg_dump** to fail and the upgrade can not be completed.
In conclusion, if the user is doing well with his postgresql-postgis configuration, it might be, from my judgment, not automatically updating these tools could be fine.

**Comment by fjperini** [comment]
@miguelangelperg PostgreSQL is by default since it is widely used. The `postgresql` update is related to `homebrew-core`. If you think that it is best to be optional, we could review this.
If you use the `postgresql` version, you will always have the latest version that causes the problem with the database, so I added the `postgresql@10` option, you should not have that problem because it remains in the same branch.

As for PostGIS I agree. It is better to have the formula with the same name so that only one of the two versions is installed and avoid conflicts.

We use our own version, because `posgis` uses `gdal` and it is better that we use `gdal2`.

Do you have both versions of both tools installed? Verify how it works only with `postgresql@10` and `posgis2` installed.

```
$ brew uninstall postgresql posgis
$ brew unlink postgresql@10 posgis2 && brew link --force postgresql@10 posgis2
```

**Comment by miguelangelperg** [comment]
Thanks @fjperini for your answers.
_Postgis_ is a very powerful tool, which is why we have approached many of them. But using it requires managing some aspects of the administration of _postgresql_ that can put in difficulty the users with less knowledge. I read: I do not want to understand _postgresql_, I want it to work.
Updates between _postgresql_ versions do not always present problems. The updates between minor versions (**10**.1 to _10_.6) do not generate any problem, but when the update is between major versions (**10**.6 to **11**) before making the upgrade it is convenient that some operations are carried out (among them update the version of _postgis_ to the more recent) .

This is the reason why I had thought that it could be a good option to avoid any update that is not essential and allow the user to decide whether to update his postgresql server.

I'll try your suggestion and I'll tell you.

I think it should work. Since _Posgis 2.2_ is compatible with _gdal2_, according to https://trac.osgeo.org/postgis/wiki/UsersWikiPostgreSQLPostGIS.

In my old postgresql10, postgis was built against gdal2. Also my previous postgresql9.6. See my output to the select postgis_ful_version () query in an old version:
_POSTGIS = " 2.4.0 r15853" PGSQL = "100" GEOS = "3.6.2-CAPI-1.10.2 4d2925d" 
PROJ = "Rel. 4.9.3, 15 de agosto de 2016" GDAL = "GDAL 2.2.2, publicado 2017/09/15" LIBXML = "2.7.8" LIBJSON = "0.12" LIBPROTOBUF = "1.2.1" RASTER_ 

My version of gdal is version 2. From my usr/local/Cellar/gdal/ I have installed versions 2.3.1_1 and 2.3.1_2, and if I ask:
`gdal-config --version`, I get: 2.3.1
`gdal-config --prefix`, I get: /usr/local/Cellar/gdal/2.3.1_2, and
`gdal-config --libs`, the result is:
-L/usr/local/Cellar/gdal/2.3.1_2/lib -lgdal

I also have _usr/local/Cellar/gdal2_ from which version 2.4.0 hangs. Maybe also built in the last upgrade.

Would we be in the same case as with postgis2?
Regards

**Comment by fjperini** [comment]
@miguelangelperg I understand what you are saying.
I think that if you only have `postgresql@10` installed, you would not have the problem when updating, since if you installed `qgis` with this option when updating it will load again the flags you used, so posgresql would not be taken into account, it will only be updated if you have it installed
I think having installed other versions of the same tool will bring conflicts, since it will use the last link. For example, with `gdal` and `gdal2`: if you run `gdal-config`, there will only be one in `/usr/local/bin` and it will be the last one linked.

For this reason it is advisable to uninstall those versions and relinkear those we use:
```

$ brew uninstall postgresql posgis gdal
$ brew unlink postgresql@10 posgis2 gdal2 && brew link --force postgresql@10 posgis2 gdal2
```

What we can do before installing or updating `qgis`: check if any of these versions are installed, and request that they be delinked or uninstalled. You could include this in the formula as a requirement.
Also if you consider that PostgreSQL is as optional.

It would be advisable to have those formulas with the same name.

The same thing also happens with PostGIS.

**Comment by fjperini** [comment]
Close these issues, we can open them later if problems arise after updating
duplicate: #813
